pennylane/ops/op_math/controlled_decompositions.py

```diff
@@ -112,3 +207,282 @@ def decomp_circuit(op):
         decomp.append(qml.RZ((omega - phi) / 2, wires=target_wire))
 
     return decomp
+
+
+def ctrl_decomp_bisect_od(
+    target_operation: typing.Union[Operator, typing.Tuple[np.ndarray, Wires]],
```

Comment:
Why are we accepting either an Operator or a tuple of matrix and wires? I would recommend limiting it to Operator for now unless we having a fairly compelling use case.

These types of permissive call signatures end causing more problem in the long run. Developers can easily cast to QubitUnitary(mat, wires=wires) in less than a single line. We don't need to make the call signature more confusing and complicated to accomodate for such a small conversion.

Reply:
this was needed to avoid instantiating unused operators in intermediate steps, which would then be recorded (incorrect)
for consistency I made all the bisect functions accept both

Reply:
see `ctrl_decomp_bisect_general` call to `ctrl_decomp_bisect_od`